src/go/k8s/Makefile

```diff
@@ -72,32 +77,39 @@ generate: controller-gen
 	$(CONTROLLER_GEN) object:headerFile="../../../licenses/boilerplate.go.txt" paths="./..."
 
 # Build the docker image
-docker-build: test
-	docker build -f Dockerfile -t ${IMG} ../
-
-# Push the docker image
-docker-push:
-	docker push ${IMG}
+docker-build:
+	docker build -f Dockerfile -t ${OPERATOR_IMG_LATEST} ../
+
+docker-tag-release:
+	docker tag ${OPERATOR_IMG_LATEST} ${OPERATOR_IMG_RELEASE}
+	docker tag ${CONFIGURATOR_IMG_LATEST} ${CONFIGURATOR_IMG_RELEASE}
+
+# Push to dockerhub
+docker-push-release:
+	if curl --silent -f -lSL https://index.docker.io/v1/repositories/${OPERATOR_IMG}/tags | grep '"name": "${TAG_NAME}"' ; then \
+	  echo "Release ${TAG_NAME} already on dockerhub, skipping push"; \
+	else \
+	  docker push ${OPERATOR_IMG_RELEASE}; \
+	  docker push ${OPERATOR_IMG_LATEST}; \
+	  docker push ${CONFIGURATOR_IMG_RELEASE}; \
+	  docker push ${CONFIGURATOR_IMG_LATEST}; \
+	fi
 
 # Build the docker image
 docker-build-configurator:
-	docker build -f cmd/configurator/Dockerfile -t ${CONFIGURATOR_IMG} ../
-
-# Push the docker image
-docker-push-configurator:
-	docker push ${CONFIGURATOR_IMG}
+	docker build -f cmd/configurator/Dockerfile -t ${CONFIGURATOR_IMG_LATEST} ../
 
 # Preload controller image to kind cluster
 push-to-kind:
-	kind load docker-image ${IMG}
-	kind load docker-image ${CONFIGURATOR_IMG}
+	kind load docker-image ${OPERATOR_IMG_LATEST}
+	kind load docker-image ${CONFIGURATOR_IMG_LATEST}
 
 # Execute end to end tests
-e2e-tests: kuttl docker-build docker-build-configurator
+e2e-tests: kuttl test docker-build docker-build-configurator
```

Comment:
a new job (`publish-images`) was introduced in the GHA workflow. Since that job starts from a fresh checkout (runs on a separate VM), there's no need to run tests again. Prior to this change, the `docker-build` target had `test` as a dependency, so it wasn't possible to invoke just the `docker-build` target, hence this change. I hope that made sense 😅

Reply:
I see, makes sense, thanks for the explanation